test/functional/ui/screen.lua

```diff
@@ -459,6 +463,25 @@ screen:redraw_debug() to show all intermediate screen states.  ]])
         end
       end
     end
+
+    if expected.extmarks ~= nil then
+      for gridid, expected_marks in pairs(expected.extmarks) do
+        local stored_marks = self._grid_win_extmarks[gridid]
+        if stored_marks == nil then
+          local str = ""
+          for _gid, _ in pairs(self._grid_win_extmarks) do
+            str = str .. tostring(_gid) .. " "
+          end
+          error(str)
+          --error('no win_extmark for grid '..tostring(gridid))
+        end
+        table.remove(self._grid_win_extmarks, gridid)
```

Comment:
you cannot remove here. We are going to need to do this check multiple times per `screen:expect()`